certification/internal/policy/operator/related_images.go

```diff
@@ -0,0 +1,89 @@
+package operator
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+
+	operatorv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/operator-framework/operator-manifest-tools/pkg/image"
+	"github.com/operator-framework/operator-manifest-tools/pkg/pullspec"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification"
+	"sigs.k8s.io/yaml"
+)
+
+type RelatedImagesCheck struct{}
+
+func (p *RelatedImagesCheck) Validate(ctx context.Context, imgRef certification.ImageReference) (bool, error) {
+	images, manifests, err := p.dataToValidate(ctx, imgRef.ImageFSPath)
+	if err != nil {
+		return false, err
+	}
+
+	return p.validate(ctx, images, manifests)
+}
+
+func (p *RelatedImagesCheck) dataToValidate(ctx context.Context, imagePath string) ([]string, map[string]struct{}, error) {
+	operatorManifests, err := pullspec.FromDirectory(filepath.Join(imagePath, "manifests"), pullspec.DefaultHeuristic)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	imageNames, err := image.Extract(operatorManifests)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	relatedImages := make(map[string]struct{}, 0)
+	for _, manifest := range operatorManifests {
+		if manifest.HasRelatedImages() {
+			csvBytes, err := manifest.ToYaml()
+			if err != nil {
+				return nil, nil, fmt.Errorf("could not get manifest yaml: %v", err)
+			}
+			var csv operatorv1alpha1.ClusterServiceVersion
+			err = yaml.Unmarshal(csvBytes, &csv)
+			if err != nil {
+				return nil, nil, fmt.Errorf("malformed CSV detected: %v", err)
+			}
+
+			for _, relatedImage := range csv.Spec.RelatedImages {
+				relatedImages[relatedImage.Image] = struct{}{}
+			}
+			break
+		}
+	}
+
+	return imageNames, relatedImages, nil
+}
+
+func (p *RelatedImagesCheck) validate(ctx context.Context, images []string, relatedImages map[string]struct{}) (bool, error) {
+	for _, image := range images {
+		if _, ok := relatedImages[image]; !ok {
+			log.Warningf("Image %s is not in relatedImages. This will eventually cause this check to fail", image)
```

Comment:
So since this is temporarily always passing and currently gets written to results, should we consider modifying the Help() or Metadata() text in a way that informs the user that this check is auto passing? So we don't get users who passed one day, and then fail when we flip the bit?